events/event_nettrace.go

```diff
@@ -15,6 +19,56 @@ var NetTraceIntegrator = func() func(basictracer.SpanEvent) {
 			tr = trace.New("tracing", t.OperationName)
 		case basictracer.EventFinish:
 			tr.Finish()
+		case basictracer.EventTag:
+			tr.LazyPrintf("%s:%v", t.Key, t.Value)
+		case basictracer.EventLogFields:
+			var buf bytes.Buffer
+
+			// Search for an "event" or "error" field for the "main" message.
+			// "event" is the key that corresponds to legacy LogEvents.
+			// "error" is the key used for error fields (opentracing-go/log.Error).
+			mainIdx := -1
+			for i, f := range t.Fields {
+				key := f.Key()
+				if key == "error" || key == "event" {
+					mainIdx = i
+					if key == "error" {
+						buf.WriteString("error: ")
+						tr.SetError()
+					}
+					fmt.Fprint(&buf, f.Value())
+					break
+				}
+			}
+
+			// If we have a "main" message, the format of the message will be:
+			//   main message (otherfield1:value, otherfield2:value, otherfield3:value)
+			// If we don't, the message will be simply:
+			//   field1:value, field2:value, field3:value
+
+			var sep string
+			if mainIdx != -1 {
+				sep = " ("
+			}
+
+			for i, f := range t.Fields {
+				if i == mainIdx {
+					continue
+				}
+				buf.WriteString(sep)
+				sep = ", "
+
+				key := f.Key()
+				if key == "error" {
```

Comment:
the OpenTracing spec infers errors from tags, not from logs... YMMV, but a single error key in a log doesn't necessarily mean the `tr` should be in an error state (IMO).